lib/react.dart

```diff
@@ -103,58 +104,204 @@ abstract class Component {
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#getdefaultprops>
   Map getDefaultProps() => {};
 
+  /// __Required.__
+  ///
+  /// When called, it should examine [props] and [state] and return a single child [Element]. This child [Element] can
+  /// be either a virtual representation of a native DOM component (such as [DivElement]) or another composite
+  /// [Component] that you've defined yourself.
+  ///
+  /// See: <https://facebook.github.io/react/docs/component-specs.html#render>
   dynamic render();
-
 }
 
-/** Synthetic event */
 
+/// A cross-browser wrapper around the browser's [nativeEvent].
+///
+/// It has the same interface as the browser's native event, including [stopPropagation] and [preventDefault], except
+/// the events work identically across all browsers.
+///
+/// See: <https://facebook.github.io/react/docs/events.html#syntheticevent>
 class SyntheticEvent {
-
+  /// Indicates whether the [Event] bubbles up through the DOM or not.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles>
   final bool bubbles;
+
+  /// Indicates whether the [Event] is cancelable or not.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable>
   final bool cancelable;
+
+  /// Identifies the current target for the event, as the [Event] traverses the DOM.
+  ///
+  /// It always refers to the [Element] the [Event] handler has been attached to as opposed to [target] which identifies
+  /// the [Element] on which the [Event] occurred.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget>
   final /*DOMEventTarget*/ currentTarget;
+
   bool _defaultPrevented;
+
   dynamic _preventDefault;
-  final dynamic stopPropagation;
+
+  /// Indicates whether or not [preventDefault] was called on the event.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented>
   bool get defaultPrevented => _defaultPrevented;
-  final num eventPhase;
-  final bool isTrusted;
-  final /*DOMEvent*/ nativeEvent;
+
+  /// Cancels the [Event] if it is [cancelable], without stopping further propagation of the event.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault>
   void preventDefault() {
     _defaultPrevented = true;
     _preventDefault();
   }
+
+  /// Prevents further propagation of the current event.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation>
+  final dynamic stopPropagation;
+
+  /// Indicates which phase of the [Event] flow is currently being evaluated.
+  ///
+  /// Possible values:
+  ///
+  /// > [Event.CAPTURING_PHASE] (1) - The [Event] is being propagated through the [target]'s ancestor objects. This
+  /// process starts with the Window, then [HtmlDocument], then the [HtmlHtmlElement], and so on through the [Element]s
```

Comment:
@trentgrover-wf I don't actually think this is a typo, even though it looks like one.

`HtmlElement` is the superclass that all `Element`s implement... whereas `HtmlHtmlElement` is literally the `<html>` DOM Node.